### PR TITLE
902: Laos Schools School Level Dashboard: ICT Facilities

### PR DESCRIPTION
**Pull request by chris-bes**
### Changes:

- Added ict facilities report to laos schools

### Screenshots:
Link in PR: https://github.com/beyondessential/tupaia-backlog/issues/902#issuecomment-692387270

### CODE REVIEW

File: packages/database/src/migrations/20200914080738-AddLaosSchoolsSchoolLevelICTFacilitiesDashboardReport-modifies-data.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    ['SchCVD012a'],
+    ['SchCVD012b'],
+    ['SchCVD013', 'SchCVD014'],
+    ['SchCVD015'],
+  ],
+  columns: ['Main', 'Extra'],
+  entityAggregation: {
+    dataSourceEntityType: 'school',
+  },
+};
```
Comment (Alastair-L) [inline packages/database/src/migrations/20200914080738-AddLaosSchoolsSchoolLevelICTFacilitiesDashboardReport-modifies-data.js:44]: This is really clean

File: packages/database/src/migrations/20200914080738-AddLaosSchoolsSchoolLevelICTFacilitiesDashboardReport-modifies-data.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      color: 'green',
+      label: '',
+      description: "**Green** indicates 'Yes' to the corresponding indicator.\n",
+    },
+    No: {
+      color: 'red',
+      label: '',
+      description: "**Red** indicates 'No' to the corresponding indicator.\n",
+    },
+    applyLocation: {
```
Comment (Alastair-L) [inline packages/database/src/migrations/20200914080738-AddLaosSchoolsSchoolLevelICTFacilitiesDashboardReport-modifies-data.js:62]: And it's nice to see that this is continuing to have use!

File: packages/database/src/migrations/20200914080738-AddLaosSchoolsSchoolLevelICTFacilitiesDashboardReport-modifies-data.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  ],
+  columns: ['Main', 'Extra'],
+  entityAggregation: {
+    dataSourceEntityType: 'school',
+  },
+};
+
+const VIEW_JSON = {
+  name: 'ICT Facilities',
+  type: 'matrix',
```
Comment (Alastair-L) [inline packages/database/src/migrations/20200914080738-AddLaosSchoolsSchoolLevelICTFacilitiesDashboardReport-modifies-data.js:48]: Is it ok as a matrix or does it need to be viewable in the sidebar? The card says "Binary Table with Response column".

File: packages/web-frontend/src/components/View/MatrixWrapper/components/HeaderRow.js [hunk 4]
```diff
@@ -88,7 +89,11 @@ export default class HeaderRow extends PureComponent {
         {columns.slice(startColumn, displayedColumnCount).map(column => (
           <div key={column.key} style={styles.headerCell}>
             <div style={styles.clinicLabel}>
-              <div style={column.isGroupHeader ? styles.facilityTypeLabel : {}}>{column.title}</div>
+              {hideColumnTitles ? null : (
+                <div style={column.isGroupHeader ? styles.facilityTypeLabel : {}}>
+                  {column.title}
+                </div>
+              )}
```
Comment (chris-bes) [inline packages/web-frontend/src/components/View/MatrixWrapper/components/HeaderRow.js:96]: I added this config `hideColumnTitles` as we dont want to show the column titles for this matrix https://github.com/beyondessential/tupaia-backlog/issues/902#issue-656270589

**Review by Alastair-L — APPROVED** [review]
Approving, but just make sure that it's ok as a matrix

File: packages/database/src/migrations/20200914080738-AddLaosSchoolsSchoolLevelICTFacilitiesDashboardReport-modifies-data.js (hunk 3 above)
Comment (chris-bes) [reply packages/database/src/migrations/20200914080738-AddLaosSchoolsSchoolLevelICTFacilitiesDashboardReport-modifies-data.js:48]: Ooopppsssss... I didn't read it carefully. Will confirm with Megan!

**Comment by chris-bes** [comment]
Just an update on this PR, Megan said that it's no longer required to remove the unused spaces. Also she wants to have a clear Grey indicator for `No data` (https://github.com/beyondessential/tupaia-backlog/issues/902#issuecomment-696476463)

File: packages/database/src/migrations/20200914080738-AddLaosSchoolsSchoolLevelICTFacilitiesDashboardReport-modifies-data.js [hunk 5]
```diff
@@ -59,6 +59,11 @@ const VIEW_JSON = {
       label: '',
       description: "**Red** indicates 'No' to the corresponding indicator.\n",
     },
+    '': {
+      color: 'grey',
+      label: '',
+      description: "**Grey** indicates 'No data' to the corresponding indicator.\n",
```
Comment (Alastair-L) [inline packages/database/src/migrations/20200914080738-AddLaosSchoolsSchoolLevelICTFacilitiesDashboardReport-modifies-data.js:65]:
```suggestion
      description: "**Grey** indicates 'No data' for the corresponding indicator.\n",
```
?

Comment (chris-bes) [reply packages/database/src/migrations/20200914080738-AddLaosSchoolsSchoolLevelICTFacilitiesDashboardReport-modifies-data.js]: whoopss. Actually I copied them from other migrations 😄

**Pull request merged**